File: frontend-web/src/services/activityService.js
/**
 * Service pour gérer l'historique des activités
 */
import apiClient from './api';

export const activityService = {
  /**
   * Lister les activités
   */
  async listActivities(page = 1, limit = 50, filters = {}) {
    const response = await apiClient.get('/activity', {
      params: {
        page,
        limit,
        ...filters,
      },
    });
    return response.data;
  },

  /**
   * Obtenir les statistiques d'activité
   */
  async getStats(days = 30) {
    const response = await apiClient.get('/activity/stats', {
      params: { days },
    });
    return response.data;
  },

  /**
   * Exporter les activités en CSV
   */
  async exportActivities(filters = {}) {
    const response = await apiClient.get('/activity/export', {
      params: filters,
      responseType: 'blob',
    });
    return response.data;
  },
};

File: mobile-app/lib/services/activity_service.dart
/// Service pour gérer l'activité
import '../services/api_service.dart';
import 'package:dio/dio.dart';

class ActivityService {
  final ApiService _apiService = ApiService();

  /// Lister les activités
  Future<Response> listActivities({
    int page = 1,
    int limit = 50,
    Map<String, dynamic>? filters,
  }) async {
    final queryParams = {
      'page': page.toString(),
      'limit': limit.toString(),
      if (filters != null) ...filters,
    };
    return await _apiService.get('/activity', queryParameters: queryParams);
  }

  /// Exporter les activités
  Future<Response> exportActivities(Map<String, dynamic>? filters) async {
    return await _apiService.post(
      '/activity/export',
      data: filters ?? {},
    );
  }

  /// Obtenir les statistiques d'activité
  Future<Response> getActivityStats() async {
    return await _apiService.get('/activity/stats');
  }
}
